.github/workflows/latex.yml

```diff
@@ -0,0 +1,50 @@
+name: Latex Build
+
+on:
+  pull_request:
+
+jobs:
+  pre-latex-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Collect diffs
+        id: "diffs"
+        run: |
+          tex=$(git --no-pager diff --name-only origin/${{ github.event.pull_request.base.ref }} -- '*.tex')
+
+          tex="${tex//'%'/'%25'}"
+          tex="${tex//$'\n'/'%0A'}"
+          tex="${tex//$'\r'/'%0D'}"
+          echo "::set-output name=tex::$tex"
+
+    outputs:
+     tex: ${{ steps.diffs.outputs.tex }}
+
+      
+  latex-build:
+    needs: [pre-latex-build]
+    if: ${{ needs.pre-latex-build.outputs.tex != '' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Compile LaTeX documents
+        uses: xu-cheng/latex-action@v2
+        with:
+          root_file: ${{ needs.pre-latex-build.outputs.tex }}
+          work_in_root_file_dir: true
+
+      - name: Pull request artifacts
+        uses: Shoeboxam/pull-request-artifacts@main
+        with:
+          commit: ${{ github.event.pull_request.head.sha }}
+          glob-pattern: rust/**/*.pdf
```

Comment:
Doing this with a glob pattern seems brittle, could pick up spurious files. You have the tex files in hand, can't you do this explicitly?